src/python/zquantum/core/interfaces/ansatz.py

```diff
@@ -0,0 +1,79 @@
+from abc import ABC, abstractmethod
+from ..circuit import Circuit
+from .ansatz_utils import ansatz_property
+from ..utils import create_symbols_map
+import copy
+import sympy
+from typing import List, Optional
+from overrides import EnforceOverrides
+import numpy as np
+
+
+class Ansatz(ABC, EnforceOverrides):
+
+    supports_parametrized_circuits = None
+    n_layers = ansatz_property("n_layers")
+
+    def __init__(self, n_layers: int):
+        """
+        Interface for implementing different ansatzes.
+        This class also caches the circuit for given ansatz parameters.
```

Comment:
Would specify here that one can use `get_executable_circuit` to get the parameterized circuit evaluated to certain parameters.